crates/rspack_plugin_css/src/dependency/import.rs

```diff
@@ -60,28 +57,18 @@ impl ModuleDependency for CssImportDependency {
   fn set_request(&mut self, request: String) {
     self.request = request;
   }
+
+  fn as_code_generatable_dependency(&self) -> Option<Box<&dyn CodeGeneratableDependency>> {
+    Some(Box::new(self))
+  }
 }
 
-impl CodeGeneratable for CssImportDependency {
-  fn generate(
+impl CodeGeneratableDependency for CssImportDependency {
+  fn apply(
     &self,
+    source: &mut CodeGeneratableSource,
     _code_generatable_context: &mut CodeGeneratableContext,
-  ) -> rspack_error::Result<CodeGeneratableResult> {
-    let mut code_gen = CodeGeneratableResult::default();
-    code_gen.visitors.push(
-      create_css_visitor!(visit_mut_stylesheet(n: &mut Stylesheet) {
-        n.rules = n
-          .rules
-          .take()
-          .into_iter()
-          .filter(|rule| match rule {
-            Rule::AtRule(at_rule) => !matches!(at_rule.prelude, Some(box AtRulePrelude::ImportPrelude(_))),
-            _ => true,
-          })
-          .collect();
-      }),
-    );
-
-    Ok(code_gen)
+  ) {
+    source.replace(self.start - 8 /* @import */, self.end, "", None);
```

Comment:
what does -8 mean here?

Reply:
because the swc bugs https://github.com/swc-project/swc/issues/7584, so here use `@import ` length to get the origin decl start